sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,343 @@
+"""
+This module can be used to solve beam bending problems in mechanics.
+
+"""
+
+from __future__ import print_function, division
+
+from sympy.printing import sstr
+from sympy.physics.mechanics import Point
+from sympy import sympify
+
+
+class Beam(object):
+
+    """
+    A Beam is a structural element that is capable of withstanding load
+    primarily by resisting against bending.
+    Beams are characterized by their profile (Second moment of area),
+    their length, and their material.
+
+    Parameters
+    ==========
+    length : Sympifyable
+        A SymPy expression representing the Beam's length.
+    elastic_modulus : Sympifyable
+        A SymPy expression representing the Beam's Modulus of Elasticity.
+        It is a measure of the stiffness of the Beam material.
+    second_moment : Sympifyable
+        A SymPy expression representing the Beam's Second moment of area.
+        It is a geometrical property of an area which reflects how its
+        points are distributed with regard to an arbitrary axis.
+
+    Examples
+    ========
+    >>> from sympy.physics.continuum_mechanics.beam import Beam
+    >>> from sympy import Symbol
+    >>> E = Symbol('E')
+    >>> I = Symbol('I')
+    >>> Beam(1, E, I)
+    Beam(1, E, I)
+
+    """
+
+    def __init__(self, length, elastic_modulus, second_moment):
+        self._length = length
+        self._elastic_modulus = elastic_modulus
+        self._second_moment = second_moment
+        self._boundary_conditions = {'deflection': [], 'moment': [], 'slope': []}
+
+    def __str__(self):
+        str_sol = 'Beam(%s, %s, %s)' % (sstr(self._length), sstr(self._elastic_modulus), sstr(self._second_moment))
+        return str_sol
+
+    __repr__ = __str__
+
+    @property
+    def length(self):
+        """Length of the Beam."""
+        return self._length
+
+    @length.setter
+    def length(self, l):
+        self._length = sympify(l)
+
+    @property
+    def elastic_modulus(self):
+        """Young's Modulus of the Beam. """
+        return self._elastic_modulus
+
+    @elastic_modulus.setter
+    def elastic_modulus(self, e):
+        self._elastic_modulus = sympify(e)
+
+    @property
+    def second_moment(self):
+        """Second moment of area of the Beam. """
+        return self._second_moment
+
+    @second_moment.setter
+    def second_moment(self, i):
+        self._second_moment = sympify(i)
+
+    def apply_boundary_conditions(self, **bcs):
```

Comment:
Since there are only three kwargs it would be better if this was:

`def apply_boundary_conditions(self, slope=None, deflection=None, moment=None)`

This also begs the question why this is needed. If you use `collections.defaultdict for list` the user could simply do something like:

`beam.boundary_conditions['moment'].append((3, 5))`